src/os/xdg.zig

```diff
@@ -193,3 +193,87 @@ test parseTerminalExec {
         try testing.expectEqualSlices([*:0]const u8, actual, &.{ "a", "-e", "b", "c" });
     }
 }
+
+/// https://specifications.freedesktop.org/basedir-spec/latest/
+pub const Dirs = enum {
+    config,
+    data,
+
+    pub fn key(self: Dirs) []const u8 {
+        return switch (self) {
+            .config => "XDG_CONFIG_DIRS",
+            .data => "XDG_DATA_DIRS",
+        };
+    }
+
+    pub fn default(self: Dirs) []const u8 {
+        return switch (self) {
+            .config => "/etc/xdg",
+            .data => "/usr/local/share:/usr/share",
+        };
+    }
+};
+
+pub const config_dirs = Dirs.config;
+pub const data_dirs = Dirs.data;
+
+pub const DirsIterator = struct {
+    data: []const u8,
+    iterator: std.mem.SplitIterator(u8, .scalar),
+
+    /// https://specifications.freedesktop.org/basedir-spec/latest/
+    pub fn init(env: std.process.EnvMap, key: Dirs) DirsIterator {
```

Comment:
EnvMap as the arg here feels wrong. EnvMap requires allocation and is also mutable in general. I think we just want the key and then to use the `getenv` machinery thereafter. To test it we can `setenv` our process.

Reply:
Now non-allocating. Zig doesn't expose `setenv` so testing resorts to bringing `setenv`/`unsetenv` through `@cImport()`.